### ISSUE TITLE
[BUG] regression after #13, cannot find file using filename

### ISSUE THREAD

**Issue by buse974** [zendframework/zend-file#19]
Hello, 
Since the merge #13 , the param $files of the method [receive($files = null)](https://github.com/zendframework/zend-file/blob/master/src/Transfer/Adapter/Http.php#L138)  does not work.

In my example code the file MyFile.txt is not moved.

``` php
$adp = new Http();
            $adp->setDestination($mydest);
            $adp->receive("MyFile.txt");
```

In method [getFiles](https://github.com/zendframework/zend-file/blob/master/src/Transfer/Adapter/AbstractAdapter.php#L1410) the name MyFile.txt is never found [(here)](https://github.com/zendframework/zend-file/blob/master/src/Transfer/Adapter/AbstractAdapter.php#L1438) because 
since the merge #13 was added "MyFile.txt" file name is rename in "phpackXwt_MyFile.txt".

**Comment by yourithielen** [comment]
I can confirm this issue, the file upload is broken.

**Comment by codisart** [comment]
Hi,

I am the author of the #13 PR.
As I noted in the comments of the pull request, the goal is to correct the behaviour of Zend_File when multiple files are uploaded with the same name in the same form as they would be moved to the same filepath and only one would be present at the end. (In some cases, if the destination folder is the same for all uploads in the app, the problem will also be present even for several upload from different users, it could be a security issue, see [#22](https://github.com/zendframework/zend-file/issues/22))

But in the case you are presenting, the problem is not my code. 

``` php
$adp = new Http();
            $adp->setDestination($mydest);
            $adp->receive("MyFile.txt");
```

If I'm not mistaking, the argument of the receive method should not be the uploaded file name but the html input name. 

``` php
$adp = new Http();
            $adp->setDestination($mydest);
            $adp->receive("input_name");
```

**Comment by codisart** [comment]
And I am mistaking !

So I just reread the code behind the receive method and in fact, it should find a file by its name.
I may be onto a fix for this issue, but I don't know what this method should be returning if they are several files with the same name in the request.

Do any one have a suggestion ?

**Comment by codisart** [comment]
Hi everyone, 

I'd really like to know what you think about this problem.
@weierophinney since you merged the original PR, maybe you got an idea ?

**Comment by sobelfallcayor** [comment]
Hi, i had this problem bat solved it to make : 
`$adapter->receive()` for all files
 or
 `$adapter->receive('one_file_name_input_html')` if there is one file on sumission.

**Comment by froschdesign** [comment]
@localheinz 
Can you look at this?

**Comment by localheinz** [comment]
Will do!

**Comment by weierophinney** [comment]
This repository has been closed and moved to laminas/laminas-file; a new issue has been opened at https://github.com/laminas/laminas-file/issues/1.
